Cargo.toml

```diff
@@ -126,3 +126,6 @@ opt-level = 3
 overflow-checks = false
 panic = 'unwind'
 rpath = false
+
+[workspace.lints.clippy]
+large_futures = "warn"
```

Comment:
I think it would help to leave some rationale here about why we have enabled this lint. Maybe something like


```suggestion
[workspace.lints.clippy]
# large stack frames cause overflows on debug builds, so try and avoid them
large_futures = "warn"
```